### PR TITLE
Fix many bugs

### LINKED ISSUE #550: Message does not update when revoking too much
When the user revokes too many reps, the message is not properly updated

### LINKED ISSUE #552: Anonymous transactions are not anonymous when user gets a notification

### LINKED ISSUE #553: Profile QuickView needs to show investor reps

### LINKED ISSUE #554: Users should start with five reps to invest upon joining repcoin

### LINKED ISSUE #555: PortfolioHeader does not show reps if it has a value of 0

### LINKED ISSUE #556: Extra collections should be wiped on repopulate or destroy database

### PR DESCRIPTION
**Pull request by ritmatter**
Fixes #556
Fixes #555
Fixes #554 
Fixes #553
Fixes #552
Fixes #550

### CODE REVIEW

File: api/handlers/transaction.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ var TransactionHandler = {
             } else {
               action = ' gave ' + amount + ' reps to you for ' + transaction.category;
             }
-            var fromText = transaction.anonymous ? 'Someone' : transaction.from.name;
```
Comment (ritmatter) [inline api/handlers/transaction.js]: Accessing the model improperly

File: api/models/User.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ var UserSchema = new Schema({
   about: {type: String, trim: true },
 
   // The reps this user has to invest
-  reps: { type: Number, required: true, default: 0 },
+  reps: { type: Number, required: true, default: 5 },
```
Comment (ritmatter) [inline api/models/User.js]: Now you start with 5 reps

File: js/components/Modal.jsx [hunk 3]
```diff
@@ -70,7 +70,7 @@ var Modal = React.createClass({
 
     // Make sure this investment has at least the amount
     if (investment.amount < amount) {
-      this.setState({ error: strings.INVESTMENT_AMOUNT_TOO_SMALL(investment.amount) });
+      this.setState({ error: true, message: strings.INVESTMENT_AMOUNT_TOO_SMALL(investment.amount) });
```
Comment (ritmatter) [inline js/components/Modal.jsx]: Not accessing state properly

File: js/components/PortfolioHeader.jsx [hunk 4]
```diff
@@ -6,7 +6,7 @@ var strings = require('../lib/strings_utils.js');
 var PortfolioHeader = React.createClass({
   render: function() {
     var repsAvailable = '';
-    if (this.props.reps) {
+    if (this.props.reps > -1) {
```
Comment (ritmatter) [inline js/components/PortfolioHeader.jsx]: Fixes the issue where when reps are 0, this.props.reps was evaluating to false

File: js/components/ProfileQuickView.jsx [hunk 5]
```diff
@@ -63,7 +63,7 @@ var ProfileQuickView = React.createClass({
 
     return (
       <div className="list-group">
-        <strong>{strings.INVESTOR_CATEGORIES}</strong>
+        <strong>{strings.INVESTOR_CATEGORIES} ( {this.state.currentUser.reps} reps )</strong>
```
Comment (ritmatter) [inline js/components/ProfileQuickView.jsx]: Show reps in the ProfileQuickView

File: scripts/destoryTestDB.sh [hunk 6]
```diff
@@ -5,5 +5,5 @@
 # ex. ./populateTestDB.sh
 
 echo "Destroying database...";
```
Comment (ritmatter) [inline scripts/destoryTestDB.sh]: Clean out the db fully on resets

**Comment by krazemon** [comment]
LGTM

**Pull request merged**